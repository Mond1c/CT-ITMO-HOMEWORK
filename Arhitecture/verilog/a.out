#! /usr/bin/vvp
:ivl_version "11.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/system.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/va_math.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/v2009.vpi";
S_0x55e1d9ccb040 .scope package, "$unit" "$unit" 2 1;
 .timescale 0 0;
S_0x55e1d9ccb1d0 .scope module, "xor2_test" "xor2_test" 3 1;
 .timescale 0 0;
v0x55e1d9cdddb0_0 .var "in1", 0 0;
v0x55e1d9cdde50_0 .var "in2", 0 0;
RS_0x7ff347b0a0d8 .resolv tri, L_0x55e1d9cde3e0, L_0x55e1d9cde4a0;
v0x55e1d9cddf20_0 .net8 "out", 0 0, RS_0x7ff347b0a0d8;  2 drivers, strength-aware
S_0x55e1d9c90cf0 .scope module, "xor2var" "xor2" 3 5, 4 1 0, S_0x55e1d9ccb1d0;
 .timescale 0 0;
    .port_info 0 /OUTPUT 1 "out";
    .port_info 1 /INPUT 1 "in1";
    .port_info 2 /INPUT 1 "in2";
L_0x55e1d9cde0c0 .functor BUFT 1, C8<771>, C4<0>, C4<0>, C4<0>;
L_0x55e1d9cde160 .functor PMOS 1, L_0x55e1d9cde0c0, v0x55e1d9cdddb0_0, C4<0>, C4<0>;
L_0x55e1d9cde020 .functor BUFT 1, C8<770>, C4<0>, C4<0>, C4<0>;
L_0x55e1d9cde2a0 .functor NMOS 1, L_0x55e1d9cde020, v0x55e1d9cdddb0_0, C4<0>, C4<0>;
L_0x55e1d9cde3e0 .functor PMOS 1, v0x55e1d9cdddb0_0, v0x55e1d9cdde50_0, C4<0>, C4<0>;
RS_0x7ff347b0a108 .resolv tri, L_0x55e1d9cde160, L_0x55e1d9cde2a0;
L_0x55e1d9cde4a0 .functor NMOS 1, RS_0x7ff347b0a108, v0x55e1d9cdde50_0, C4<0>, C4<0>;
v0x55e1d9c90ed0_0 .net8 "GRD", 0 0, L_0x55e1d9cde020;  1 drivers, strength-aware
v0x55e1d9cdd940_0 .net8 "PWR", 0 0, L_0x55e1d9cde0c0;  1 drivers, strength-aware
v0x55e1d9cdda00_0 .net "in1", 0 0, v0x55e1d9cdddb0_0;  1 drivers
v0x55e1d9cddaa0_0 .net "in2", 0 0, v0x55e1d9cdde50_0;  1 drivers
v0x55e1d9cddb60_0 .net8 "out", 0 0, RS_0x7ff347b0a0d8;  alias, 2 drivers, strength-aware
v0x55e1d9cddc70_0 .net8 "w", 0 0, RS_0x7ff347b0a108;  2 drivers, strength-aware
    .scope S_0x55e1d9ccb1d0;
T_0 ;
    %vpi_call/w 3 8 "$monitor", "%0t: in1: %b, in2: %b, out = %b", $time, v0x55e1d9cdddb0_0, v0x55e1d9cdde50_0, v0x55e1d9cddf20_0 {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55e1d9cdddb0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55e1d9cdde50_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55e1d9cdddb0_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x55e1d9cdde50_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x55e1d9cdddb0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55e1d9cdde50_0, 0, 1;
    %delay 1, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x55e1d9cdddb0_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x55e1d9cdde50_0, 0, 1;
    %end;
    .thread T_0;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "-";
    "test_xor.sv";
    "xor.sv";
