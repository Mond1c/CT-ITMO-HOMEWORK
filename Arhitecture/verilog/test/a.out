#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "C:\Users\pocht\scoop\apps\iverilog\current\lib\ivl\system.vpi";
:vpi_module "C:\Users\pocht\scoop\apps\iverilog\current\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\Users\pocht\scoop\apps\iverilog\current\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\Users\pocht\scoop\apps\iverilog\current\lib\ivl\v2005_math.vpi";
:vpi_module "C:\Users\pocht\scoop\apps\iverilog\current\lib\ivl\va_math.vpi";
:vpi_module "C:\Users\pocht\scoop\apps\iverilog\current\lib\ivl\v2009.vpi";
S_000001eaa8b55f80 .scope package, "$unit" "$unit" 2 1;
 .timescale 0 0;
S_000001eaa8b56110 .scope package, "test" "test" 3 1;
 .timescale 0 0;
C000001eaa8b56e20  .class "CPU" [1]
   0: "number", "sb32"
 ;
S_000001eaa8b51fc0 .scope class, "CPU" "CPU" 3 2, 3 2 0, S_000001eaa8b56110;
 .timescale 0 0;
S_000001eaa8b52150 .scope autofunction.void, "display" "display" 3 10, 3 10 0, S_000001eaa8b51fc0;
 .timescale 0 0;
v000001eaa8b522e0_0 .var/cobj "@";
TD_test.CPU.display ;
    %load/obj v000001eaa8b522e0_0;
    %prop/v 0;
    %pop/obj 1, 0;
    %vpi_call/w 3 11 "$display", "This is CPU %d", S<0,vec4,s32> {1 0 0};
    %end;
S_000001eaa8b52380 .scope autofunction.obj, "new" "new" 3 5, 3 5 0, S_000001eaa8b51fc0;
 .timescale 0 0;
v000001eaa8b52510_0 .var/cobj "@";
v000001eaa8b525b0_0 .var/2s "number", 31 0;
TD_test.CPU.new ;
    %load/obj v000001eaa8b52510_0;
    %prop/v 0;
    %pop/obj 1, 0;
    %load/obj v000001eaa8b52510_0;
    %store/prop/v 0, 32; Store in bool property number
    %pop/obj 1, 0;
    %end;
S_000001eaa89d9bf0 .scope module, "ok" "ok" 4 4;
 .timescale 0 0;
v000001eaa8b52880_0 .var/cobj "cpu";
S_000001eaa8b52650 .scope begin, "$unm_blk_1" "$unm_blk_1" 4 7, 4 7 0, S_000001eaa89d9bf0;
 .timescale 0 0;
v000001eaa8b527e0_0 .var/2u "p", 63 0;
    .scope S_000001eaa89d9bf0;
T_2 ;
    %fork t_1, S_000001eaa8b52650;
    %jmp t_0;
    .scope S_000001eaa8b52650;
t_1 ;
    %pushi/vec4 1, 0, 32;
    %ix/load 4, 32, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v000001eaa8b527e0_0, 4, 32;
    %pushi/vec4 2, 0, 32;
    %ix/load 4, 0, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v000001eaa8b527e0_0, 4, 32;
    %load/vec4 v000001eaa8b527e0_0;
    %parti/u 32, 32, 32;
    %load/vec4 v000001eaa8b527e0_0;
    %parti/u 32, 0, 32;
    %vpi_call/w 4 11 "$display", "%d %d", S<1,vec4,s32>, S<0,vec4,s32> {2 0 0};
    %alloc S_000001eaa8b52380;
    %new/cobj C000001eaa8b56e20;
    %pushi/vec4 1, 0, 32;
    %store/vec4 v000001eaa8b525b0_0, 0, 32;
    %store/obj v000001eaa8b52510_0;
    %callf/obj TD_test.CPU.new, S_000001eaa8b52380;
    %load/obj v000001eaa8b52510_0;
    %free S_000001eaa8b52380;
    %store/obj v000001eaa8b52880_0;
    %alloc S_000001eaa8b52150;
    %load/obj v000001eaa8b52880_0;
    %store/obj v000001eaa8b522e0_0;
    %callf/void TD_test.CPU.display, S_000001eaa8b52150;
    %free S_000001eaa8b52150;
    %end;
    .scope S_000001eaa89d9bf0;
t_0 %join;
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "-";
    "./pack.sv";
    ".\test.sv";
